app/jobs/benchmark_regression_job.rb

```diff
@@ -0,0 +1,46 @@
+require 'uri'
+require 'net/http'
+require 'json'
+
+class BenchmarkRegressionJob < ActiveJob::Base
+  queue_as :default
+
+  def perform(benchmark_run_id)
+
+    benchmark_run = BenchmarkRun.find_by_id(benchmark_run_id)
+
+    category = benchmark_run.result.keys[0]
+
+    previous_benchmark_runs = BenchmarkRun.where('created_at < ?', benchmark_run.created_at).where(initiator_type: Commit, benchmark_type: benchmark_run.benchmark_type,
```

Comment:
> Looking at the model of benchmark_runs above query should fetch the required results.

Can you explain in your own words what we are trying to query for here? I still don't you understand what we need to be fetching here :crying_cat_face: 

> As we are giving benchmark type as a parameter why do we need to check for the right repository?
I didn't say anything about checking :disappointed: The emphasis is to let you know that you have to ensure that the benchmark_runs are scope to the right repository.